prometheus-node-exporter-lua remove zero values

NoctivagusObitus opened the pull request:
depending on the configuration there may be multiple interfaces creating multiple time series always
reporting 0 value. omiting them from the export saves resources. most notably cpu.

- remove time series with value of zero
- make 'omit_zero_values' configurable
- formating accoding to shfmt and stylua

Signed-off-by: Markus Hube <markus.hube@gmail.com>

NoctivagusObitus commented:
@champtar may I kindly ask if you could have a look at this? I would be happy to change the pull request given you have concerns.

rwrnet commented:
@champtar has this any chance to be merged? Like to have this option too.

NoctivagusObitus commented:
@1715173329 do you think we may get this merged? I would be happy to rework the pr as offered [here](https://github.com/openwrt/packages/issues/22946#issuecomment-2440763843) if so desired.

1715173329 commented:
I'm neither the maintainer of this package nor familiar with it, sorry for that.
But at least, please bump PKG_RELEASE, and split your commits. One commit for adding omit_zero_values option, and another for style fixes.

@champtar kindly ping

NoctivagusObitus commented:
Sure. I am aware that your are not the maintainer and that you will not do the review or merge. Still thank you for your fast response.

I have split the changes into two commits and implemented the "only counter" logic.

scMarkus commented:
any chance on progressing on this?

NoctivagusObitus commented:
@1715173329 sorry for bothering you again. might there by any chance of progressing without waiting for @champtar to respond?

1715173329 commented:
Please rebase this PR to resolve conflicts, otherwise LGTM.

NoctivagusObitus commented:
Thanks a lot. since there have been some changes in place I touched I would like to rebase and let it run on my devices for a bit to make sure I did not mess up anything. I will get back to you.

1715173329 commented:
Merged, thank you!
